packages/filesystem/src/browser/file-dialog/file-dialog.ts

```diff
@@ -218,3 +275,86 @@ export class FileDialog extends AbstractDialog<MaybeArray<FileStatNode>> {
     }
 
 }
+
+@injectable()
+export class SaveFileDialog extends FileDialog<string> {
+
+    fileNameField: HTMLInputElement;
+
+    constructor(
+        @inject(SaveFileDialogProps) readonly props: SaveFileDialogProps,
+        @inject(FileDialogWidget) readonly widget: FileDialogWidget
+    ) {
+        super(props, widget);
+        widget.addClass(SAVE_DIALOG_CLASS);
+    }
+
+    protected getAcceptButtonLabel(): string {
+        return this.props.saveLabel ? this.props.saveLabel : 'Save';
+    }
+
+    protected onUpdateRequest(msg: Message): void {
+        // Update file name field when changing a selection
+        if (this.widget.model.selectedFileStatNodes.length === 1) {
+            const fileStat = this.widget.model.selectedFileStatNodes[0];
+            if (!fileStat.fileStat.isDirectory) {
+                this.fileNameField.value = fileStat.name;
+            }
+        } else {
+            this.fileNameField.value = '';
+        }
+
+        // Continue updating the dialog
+        super.onUpdateRequest(msg);
+    }
+
+    isValid(value: string): string {
+        if (this.fileNameField.value) {
+            return '';
+        }
+
+        // Returning a space disables the 'Save' button and doesn't show an error message at the bottom
+        return ' ';
+    }
+
+    get value(): string {
+        if (this.widget.model.selectedFileStatNodes.length === 1) {
+            const node = this.widget.model.selectedFileStatNodes[0];
+
+            if (node.fileStat.isDirectory) {
+                const path = node.uri.path.toString();
+                return path + '/' + this.fileNameField.value;
+            }
+
+            const parent = node.uri.parent.path.toString();
```

Comment:
`return node.uri.parent.resolve(this.fileNameField.value)`

Reply:
I will try, thanks!
```resolve(path: string | Path): URI``` doesn't have an explanation comment, it's not clear what this method does.

Reply:
Reworked. Thanks!